typedconfig/casts.py

```diff
@@ -0,0 +1,11 @@
+from enum import Enum, EnumMeta
+from typing import  Callable
+
+def enum_cast(enum_type: EnumMeta)->Callable[[str], Enum]:
+    def getter(name: str)->Enum:
```

Comment:
The type signature on this won't give nice autocompletion as it is, since it just returns a generic `Enum` so whatever uses `enum_cast` will get autocomplete for things like `.name` and `.value` but not for the specific enum type. This can be fixed like this:
```python
from typing import TypeVar, Callable, Type

TEnum = TypeVar('TEnum', bound="Enum")


def enum_cast(enum_type: Type[TEnum]) -> Callable[[str], TEnum]:
    def getter(name: str) -> TEnum:
        ...
```

Reply:
This still isn't quite right - return type should be `Callable[[str], TEnum]`